test/test_paths.py

```diff
@@ -140,3 +140,13 @@ def okgoogle(url):
   assert path.bucket == '/tmp'
   assert path.dataset == 'removeme'
   assert path.layer == 'layer'
+
+
+def test_windows_path_extraction():
+  """ This test is added specifically to make sure
+  that no_bucket_basepath does not contain the bucket,
+  for windows paths. """
+  extract = paths.extract(r'file://C:\wow\this\is\a\cool\path', windows=True, disable_toabs=True)
+  assert extract.protocol == 'file'
+  assert extract.bucket == 'C:\\wow\\'
```

Comment:
Hmm, I'm not sure automated testing will give the correct result because `split_char` is set by `os.path.sep`.

Reply:
Maybe it's worth redefining this part of the code:

```  split_char = '/'
  if protocol == 'file':
    cloudpath = abspath(cloudpath)
    if windows:
      bucket_re = windows_file_re
      split_char = '\\'```

Reply:
Hmm, I'm not sure I follow your point. 

That part of the test is actually from the other test that already existed in this file. I only added the line:
```python
assert 'wow' not in extract.no_bucket_basepath
```
everything else was copy paste of the top of the previous test.

In any case the tests pass on linux and windows platforms, so I'm not sure why it wouldn't pass automated testing. The split_char is set to `'\\'` in the test because `windows=True` in:
```python
extract = paths.extract(r'file://C:\wow\this\is\a\cool\path', windows=True, disable_toabs=True)
```
Maybe there is something I don't understand about the automated testing. Can you explain what you mean?